### PR TITLE
feat: New Analytics/Enrollment API EntryPoint[2.41-DHIS2-15291_3]

### PR DESCRIPTION
**Pull request by d-bernat**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-administration/src/test/java/org/hisp/dhis/jsonschema/JsonSchemaValidatorTest.java [hunk 1]
```diff
@@ -36,15 +36,22 @@
...
 class JsonSchemaValidatorTest {
 
   private final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
 
+  @BeforeAll()
+  static void setup() {
+    Locale.setDefault(new Locale("en", "US"));
+  }
+
   @Test
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-administration/src/test/java/org/hisp/dhis/jsonschema/JsonSchemaValidatorTest.java:55]: ah! good one! 😄

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java [hunk 2]
```diff
@@ -136,6 +141,12 @@ public abstract class AbstractJdbcEventAnalyticsManager {
 
   private static final String COL_VALUE = "value";
 
+  private static final String COL_PERIOD = "pe";
+
+  private static final String COL_ORGANIZATION = "ou";
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java:146]: I'm pretty sure we already have these defined somewhere else. Does it still makes sense to duplicate here ?

**Review by gnespolino — APPROVED** [review]
good job, left some minor comments

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java [hunk 3]
```diff
@@ -157,19 +166,25 @@ protected Grid createGridWithHeaders(EventQueryParams params) {
...
+    if (params.getEndpointAction() == RequestTypeAware.EndpointAction.AGGREGATE) {
+      paramsList = queryPlanner.planAggregateQuery(params);
+    } else {
+      paramsList = List.of(queryPlanner.planEnrollmentQuery(params));
+    }
 
     long count = 0;
+    for (EventQueryParams queryParams : paramsList) {
+      timer.getSplitTime("Planned event query, got partitions: " + queryParams.getPartitions());
+      if (queryParams.isTotalPages()) {
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java:180]: I'm not sure if we should be counting and handling pagination as this is aggregated data only. But this is part of our discussion regarding the renaming of the method, let's see...

**Review by maikelarabori — COMMENTED** [review]
Looks good Dusan, especially the controller and tests. The methods/operations are well covered.

I left a few comments, please have a look.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java:146]: done

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/EventDataQueryRequest.java [hunk 4]
```diff
@@ -339,7 +339,8 @@ public EventDataQueryRequestBuilder fromCriteria(EnrollmentAnalyticsQueryCriteri
       }
 
       Set<String> dimensions;
-      if (criteria.isQueryEndpoint()) {
+      if (criteria.isQueryEndpoint()
+          || (criteria.isAggregateEndpoint() && criteria.isEnrollmentEndpointItem())) {
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/EventDataQueryRequest.java:343]: done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java [hunk 5]
```diff
@@ -129,6 +131,13 @@ protected Grid getGrid(EventQueryParams params) {
             .flatMap(dk -> dk.getKeywords().stream())
             .collect(toList());
 
+    List<DimensionalObject> periods =
+        params.isAggregatedEnrollments()
+            ? params.getDimensions().stream()
+                .filter(d -> d.getDimensionType() == DimensionType.PERIOD)
+                .toList()
+            : new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java]: done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java [hunk 6]
```diff
@@ -129,6 +131,13 @@ protected Grid getGrid(EventQueryParams params) {
             .flatMap(dk -> dk.getKeywords().stream())
             .collect(toList());
 
+    List<DimensionalObject> periods =
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java]: done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java [hunk 7]
```diff
@@ -157,19 +166,25 @@ protected Grid createGridWithHeaders(EventQueryParams params) {
   protected long addEventData(Grid grid, EventQueryParams params) {
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java]: done (addData)

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java (hunk 3 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java]: done

**Review by maikelarabori — APPROVED** [review]
Looks good! :+1:

**Pull request merged**